helpers/elixir/bin/check_update.exs

```diff
@@ -1,31 +1,69 @@
-[dependency_name] = System.argv()
+defmodule UpdateChecker do
+  def run(dependency_name, credentials) do
+    set_credentials(credentials)
 
-# dependency atom
-dependency = String.to_atom(dependency_name)
+    # Update the lockfile in a session that we can time out
+    task = Task.async(fn -> do_resolution(dependency_name) end)
+    Task.await(task, 30000)
 
-# Fetch dependencies that needs updating
-{dependency_lock, rest_lock} = Map.split(Mix.Dep.Lock.read(), [dependency])
+    # Read the new lock
+    {updated_lock, _updated_rest_lock} =
+      Map.split(Mix.Dep.Lock.read(), [String.to_atom(dependency_name)])
 
-try do
-  Mix.Dep.Fetcher.by_name([dependency_name], dependency_lock, rest_lock, [])
-
-  # Check the dependency version in the new lock
-  {updated_lock, _updated_rest_lock} = Map.split(Mix.Dep.Lock.read(), [dependency])
-
-  version =
+    # Get the new dependency version
     updated_lock
-    |> Map.get(dependency)
+    |> Map.get(String.to_atom(dependency_name))
     |> elem(2)
+  end
+
+  defp set_credentials(credentials) do
+    credentials
+    |> Enum.reduce([], fn cred, acc ->
```

Comment:
I'm not sure to understand the body of this function, can you give me more details please?

Reply:
It's doing the equivalent of looping through `mix hex.organization auth <org_name> --key <key>` for each credentials. It starts out with an array that looks like `[org, key, org2, key2]`, groups them into pairs, and then does a second pass to set them.

Reply:
Clearly pretty inelegant - would be very open to suggestions of a better way!

Reply:
```elixir
credentials
|> Enum.zip()
|> Enum.each(fn {organisation, token} ->
      hexpm = Hex.Repo.get_repo("hexpm")
      
      repo = %{
        url: hexpm.url <> "/repos/#{organization}",
        public_key: nil,
        auth_key: token
      }

      Hex.Config.read()
      |> Hex.Config.read_repos()
      |> Map.put("hexpm:#{organization}", repo)
      |> Hex.Config.update_repos()
end)
```

Reply:
@greysteil here's the refactoring :)

Reply:
You are a hero @stevedomin 🎉

Reply:
Hmmm, actually, that's not working for me - mind having a try getting it running with the specs in `spec/dependabot/update_checkers/elixir/hex_spec.rb`?

Reply:
Ah yeah my bad, British spelling instead of US.